includes/class-wc-form-handler.php

```diff
@@ -205,12 +206,18 @@ public static function save_account_details() {
 		$user->last_name    = $account_last_name;
 
 		// Prevent emails being displayed, or leave alone.
-		$user->display_name = is_email( $current_user->display_name ) ? $user->first_name : $current_user->display_name;
-
+		$user->display_name = is_email( $current_user->display_name ) ? sprintf( __( '%1$s %2$s', 'display name', 'woocommerce' ), $user->first_name, $user->last_name ) : $account_display_name;
```

Comment:
Let's try moving this email check to `set_display_name` method in the `WC_Customer`. Then we can handle it everywhere in the CRUD and prevent email ever being shown.